sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClient.java

```diff
@@ -1565,11 +1569,9 @@ private boolean isADFSTenant() {
 
     private InputStream getCertificateInputStream() throws IOException {
         if (certificatePath != null) {
-            return new FileInputStream(certificatePath);
-        } else if (certificate != null) {
-            return certificate;
+            return new BufferedInputStream(new FileInputStream(certificatePath));
         } else {
-            return null;
+            return certificate;
```

Comment:
If certificate is null we return null anyways, so just return certificate no matter what